sympy/stats/crv_types.py

```diff
@@ -1563,58 +1563,78 @@ def GammaInverse(name, a, b):
     return rv(name, GammaInverseDistribution, (a, b))
 
 #-------------------------------------------------------------------------------
-# Gumbel distribution --------------------------------------------------------
+# Gumbel distribution (Maximum and Minimum) --------------------------------------------------------
 
 
 class GumbelDistribution(SingleContinuousDistribution):
-    _argnames = ('beta', 'mu')
+    _argnames = ('beta', 'mu', 'minimum')
 
     set = Interval(-oo, oo)
 
     @staticmethod
-    def check(beta, mu):
+    def check(beta, mu, minimum):
         _value_check(beta > 0, "Scale parameter beta must be positive.")
 
     def pdf(self, x):
         beta, mu = self.beta, self.mu
         z = (x - mu)/beta
-        return (1/beta)*exp(-(z + exp(-z)))
+        f_max = (1/beta)*exp(-z - exp(-z))
+        f_min = (1/beta)*exp(z - exp(z))
+        return Piecewise((f_min, self.minimum), (f_max, not self.minimum))
 
     def _cdf(self, x):
         beta, mu = self.beta, self.mu
-        return exp(-exp((mu - x)/beta))
+        z = (x - mu)/beta
+        F_max = exp(-exp(-z))
+        F_min = 1 - exp(-exp(z))
+        return Piecewise((F_min, self.minimum), (F_max, not self.minimum))
 
     def _characteristic_function(self, t):
-        return gamma(1 - I*self.beta*t) * exp(I*self.mu*t)
+        cf_max = gamma(1 - I*self.beta*t) * exp(I*self.mu*t)
+        cf_min = gamma(1 + I*self.beta*t) * exp(I*self.mu*t)
+        return Piecewise((cf_min, self.minimum), (cf_max, not self.minimum))
 
     def _moment_generating_function(self, t):
-        return gamma(1 - self.beta*t) * exp(self.mu*t)
+        mgf_max = gamma(1 - self.beta*t) * exp(self.mu*t)
+        mgf_min = gamma(1 + self.beta*t) * exp(self.mu*t)
+        return Piecewise((mgf_min, self.minimum), (mgf_max, not self.minimum))
 
-def Gumbel(name, beta, mu):
+def Gumbel(name, beta, mu, minimum=False):
     r"""
     Create a Continuous Random Variable with Gumbel distribution.
 
     The density of the Gumbel distribution is given by
 
+    For Maximum
+
     .. math::
         f(x) := \dfrac{1}{\beta} \exp \left( -\dfrac{x-\mu}{\beta}
                 - \exp \left( -\dfrac{x - \mu}{\beta} \right) \right)
 
     with :math:`x \in [ - \infty, \infty ]`.
 
+    For Minimum
+
+    .. math::
+        f(x) := \frac{e^{- e^{\frac{- \mu + x}{\beta}} + \frac{- \mu + x}{\beta}}}{\beta}
+
+    with :math:`x \in [ - \infty, \infty ]`.
+
     Parameters
     ==========
 
-    mu: Real number, 'mu' is a location
-    beta: Real number, 'beta > 0' is a scale
+    mu : Real number, 'mu' is a location
+    beta : Real number, 'beta > 0' is a scale
+    minimum : Boolean, by default, False, for enabling minimum distribution
```

Comment:
The default corresponds to maximum distribution by your setting.

Reply:
Is it needed to be changed to minimum distribution by default? AFAIK, the original tests pass on keeping the maximum distribution as default. If minimum is set to True then they will fail. Am I wrong somewhere?

Reply:
>    minimum : Boolean, by default, False, for enabling minimum distribution

This looks a bit misleading. It reads as if the default is for enabling minimum distribution.

Reply:
I have made the change please take a look if now its clear. Resolve this conversation if you are in agreement of the change. Thanks